coffeecard/CoffeeCard.WebApi/Controllers/v2/ProductsController.cs

```diff
@@ -0,0 +1,93 @@
+
+    public class ProductsController : ControllerBase
+    {
+        private readonly ClaimsUtilities _claimsUtilities;
+        private readonly IMapperService _mapperService;
+        private readonly IProductService _productService;
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProductsController"/> class.
+        /// </summary>
+        public ProductsController(IProductService productService, IMapperService mapper,
+            ClaimsUtilities claimsUtilities)
+        {
+            _productService = productService;
+            _mapperService = mapper;
+            _claimsUtilities = claimsUtilities;
+        }
+
+        /// <summary>
+        /// Adds a new product to the database.
+        /// </summary>
+        /// <param name="initiateProductRequest">The request containing the details of the product to be added and allowed user groups.</param>
+        /// <returns> The newly added product wrapped in a InitiateProductResponse object.</returns>
+        /// <response code="200">The request was successful, and the product was added.</response>
+        [HttpPost("add")]
+        [AuthorizeRoles(UserGroup.Board)]
+        [ProducesResponseType(typeof(InitiateProductResponse), StatusCodes.Status200OK)]
+        public async Task<IActionResult> AddProduct(InitiateProductRequestWithUserGroups initiateProductRequest)
+        {
+            return Ok(await _productService.AddProduct(initiateProductRequest.Product, initiateProductRequest.AllowedUserGroups));
+        }
+        
+        
+        /// <summary>
+        /// Updates a product with the specified changes.
+        /// </summary>
+        /// <param name="product">The request containing the changes to be applied to the product.</param>
+        /// <returns>A response indicating the result of the update operation.</returns>
+        /// <response code="200">The request was successful, and the product was updated.</response>
+        [HttpGet("update")]
+        [AuthorizeRoles(UserGroup.Board)]
+        [ProducesResponseType(typeof(InitiateProductResponse), StatusCodes.Status200OK)]
+        public async Task<IActionResult> UpdateProduct(InitiateProductRequest product)
+        {
+            return Ok(await _productService.UpdateProduct(product));
+        }
+        
+        
+        /// <summary>
+        /// Deactivates a product by setting its visibility to false.
+        /// </summary>
+        /// <param name="productId">The ID of the product to be deactivated.</param>
+        /// <returns>A response indicating the result of the deactivation operation.</returns>
+        /// <response code="200">The request was successful, and the product was deactivated.</response>
+        /// <response code="404">The product with the specified ID was not found.</response>
+        [HttpPost("deactivate/{productId}")]
+        [ProducesResponseType(typeof(void), StatusCodes.Status200OK)]
```

Comment:
This needs to be limited to the board as well

Reply:
We might also want to simply use the updateProduct endpoint instead when we change the visibility. Thoughts @jonasanker ?